File: POH_V7/Ruby/c1.rb
n = gets.to_i
m1 = gets.to_i
x = gets.chomp.split(" ")
x.map!(&:to_i)
m2 = gets.to_i
y = gets.chomp.split(" ")
y.map!(&:to_i)
buy = Array.new()
for i in y
  buy.push(i) if x.index(i) == nil
end
buy.sort!
if buy.size == 0 then
  puts "None"
else
  f = false
  for n in buy
    print " " if f == true
    print n
    f = true
  end
  puts ""
end

File: POH_V7/Python/c1.py
n = int(input())
m = int(input())
x = [ int(i) for i in input().split() ]
m = int(input())
y = [ int(i) for i in input().split() ]
buy = [ i for i in y if i not in x ]
if len(buy) == 0:
    print("None")
else:
    buy.sort()
    for n in buy:
        if n != buy[0]:
            print(" ",end="")
        print(n,end="")
    print("")
